### ISSUE TITLE
Code uploading on DUE is not working in UBUNTU

### ISSUE THREAD

**Issue by Atirma** [arduino/Arduino#3923]
Hi,
We're facing exactly the problem declared in this link.
http://askubuntu.com/questions/681555/ubuntu-14-04-arduino-due-upload-firmware-issue
This problem is serious as it renders the DUE board useless under UBUNTU.

**Comment by NicoHood** [comment]
Why dont you switch to the 16u2 usb port?

**Comment by ffissore** [comment]
@Fede85 is having the same issue with his ubuntu 14.04, while I'm not with mine. My only guess so far it's that it's hardware related. Can you spare a 14.04 on a different pc and retest?

**Comment by Atirma** [comment]
Hi ffissore,
We have tested this issues on two PCs. Luckily, one exhibits this problem, the other doesn't.
Using command 'lsb_release --all' we obtain the following with PC that is able of uploading to the DUE:
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.1 LTS
Release:        14.04
Codename:       trusty

Executing the same command with the PC where bossac fails, we get:

No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.3 LTS
Release:        14.04
Codename:       trusty

Main difference is that the first one is Ubuntu 14.04.1 LTS  while the second one is Ubuntu 14.04.3 LTS

'uname -a' returns:

3.13.0-32-generic #57-Ubuntu SMP Tue Jul 15 03:51:08 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux for Ubuntu 14.04.1 LTS, and

3.13.0-65-generic #105-Ubuntu SMP Mon Sep 21 18:50:58 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux for Ubuntu 14.04.3 LTS

I'm attaching also a couple of log files detailing the modules loaded in both cases.
Regards,


------- For the PC where bossac works ---------------------------------------------------------------------------
Script iniciado (miÃ© 07 oct 2015 12:24:41 WEST
)[38;5;202m[38;5;245mantodom[00m@[38;5;172mdis172ac[00m:[38;5;5m~[00mâŸ« vim EKD_time.tex exit[Kgit push origin mastercommit -a -m "cambios"status[Kcd git-projects/siani/
[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[Cdmesg[Kmake upload[6Pdmesgcd git-projects/siani/
[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[12Pgit statuscommit -a -m "cambios"[4Ppush origin master
[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[Cexit[Kvim EKD_time.tex [Klsmod
Module                  Size  Used by
cdc_acm                28803  0 
ipt_MASQUERADE         12880  3 
iptable_nat            13011  1 
nf_nat_ipv4            13263  1 iptable_nat
nf_nat                 21841  3 ipt_MASQUERADE,nf_nat_ipv4,iptable_nat
nf_conntrack_ipv4      15012  2 
nf_defrag_ipv4         12758  1 nf_conntrack_ipv4
xt_conntrack           12760  1 
nf_conntrack           96976  6 ipt_MASQUERADE,nf_nat,nf_nat_ipv4,xt_conntrack,iptable_nat,nf_conntrack_ipv4
ipt_REJECT             12541  2 
xt_CHECKSUM            12549  1 
iptable_mangle         12695  1 
xt_tcpudp              12884  6 
bridge                110793  0 
stp                    12976  1 bridge
llc                    14552  2 stp,bridge
ip6table_filter        12815  0 
ip6_tables             27025  1 ip6table_filter
iptable_filter         12810  1 
ip_tables              27239  3 iptable_filter,iptable_mangle,iptable_nat
ebtable_nat            12807  0 
ebtables               30913  1 ebtable_nat
x_tables               34059  11 ip6table_filter,xt_CHECKSUM,ip_tables,xt_tcpudp,ipt_MASQUERADE,xt_conntrack,iptable_filter,ebtables,ipt_REJECT,iptable_mangle,ip6_tables
bnep                   19624  2 
rfcomm                 69160  0 
bluetooth             391196  10 bnep,rfcomm
binfmt_misc            17468  1 
nls_iso8859_1          12713  1 
snd_hda_codec_hdmi     46254  4 
joydev                 17381  0 
snd_hda_codec_via      27860  1 
intel_rapl             18773  0 
x86_pkg_temp_thermal    14205  0 
intel_powerclamp       14705  0 
coretemp               13435  0 
kvm_intel             143060  0 
kvm                   451511  1 kvm_intel
crct10dif_pclmul       14289  0 
crc32_pclmul           13113  0 
ghash_clmulni_intel    13216  0 
dm_multipath           22873  0 
scsi_dh                14882  1 dm_multipath
snd_hda_intel          52355  7 
snd_hda_codec         192906  3 snd_hda_codec_hdmi,snd_hda_codec_via,snd_hda_intel
aesni_intel            55624  0 
nouveau              1097199  4 
aes_x86_64             17131  1 aesni_intel
ppdev                  17671  0 
lrw                    13286  1 aesni_intel
gf128mul               14951  1 lrw
glue_helper            13990  1 aesni_intel
ablk_helper            13597  1 aesni_intel
cryptd                 20359  3 ghash_clmulni_intel,aesni_intel,ablk_helper
snd_hwdep              13602  1 snd_hda_codec
lp                     17759  0 
mxm_wmi                13021  1 nouveau
wmi                    19177  2 mxm_wmi,nouveau
serio_raw              13462  0 
snd_pcm               102099  4 snd_hda_codec_hdmi,snd_hda_codec,snd_hda_intel
ttm                    85115  1 nouveau
drm_kms_helper         53081  1 nouveau
snd_page_alloc         18710  2 snd_pcm,snd_hda_intel
drm                   303102  6 ttm,drm_kms_helper,nouveau
snd_timer              29482  1 snd_pcm
snd                    69238  20 snd_hwdep,snd_timer,snd_hda_codec_hdmi,snd_hda_codec_via,snd_pcm,snd_hda_codec,snd_hda_intel
i2c_algo_bit           13413  1 nouveau
mei_me                 18627  0 
lpc_ich                21080  0 
mei                    82276  1 mei_me
soundcore              12680  1 snd
shpchp                 37032  0 
parport_pc             32701  1 
parport                42348  3 lp,ppdev,parport_pc
video                  19476  1 nouveau
mac_hid                13205  0 
hid_generic            12548  0 
usbhid                 52570  0 
psmouse               106678  0 
hid                   106148  2 hid_generic,usbhid
pata_acpi              13038  0 
atl1c                  46086  0 
usb_storage            62209  1 
[38;5;202m[38;5;245mantodom[00m@[38;5;172mdis172ac[00m:[38;5;5m~[00mâŸ« dpkt-[K -[K[K[K[K[K[Kdmesg | tail -n 10[K[K25
[1245712.206747] systemd-hostnamed[15981]: Warning: nss-myhostname is not installed. Changing the local hostname might make it unresolveable. Please install nss-myhostname!
[1248845.977620] systemd-hostnamed[21113]: Warning: nss-myhostname is not installed. Changing the local hostname might make it unresolveable. Please install nss-myhostname!
[1250687.436145] usb 1-1.3: USB disconnect, device number 24
[1292617.816446] audit_printk_skb: 60 callbacks suppressed
[1292617.816451] type=1400 audit(1444197200.174:306): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="/usr/lib/cups/backend/cups-pdf" pid=24150 comm="apparmor_parser"
[1292617.816460] type=1400 audit(1444197200.174:307): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="/usr/sbin/cupsd" pid=24150 comm="apparmor_parser"
[1292617.817037] type=1400 audit(1444197200.174:308): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="/usr/sbin/cupsd" pid=24150 comm="apparmor_parser"
[1299159.967777] usb 1-1.3: new high-speed USB device number 25 using ehci-pci
[1299160.060046] usb 1-1.3: New USB device found, idVendor=0424, idProduct=2504
[1299160.060051] usb 1-1.3: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[1299160.060331] hub 1-1.3:1.0: USB hub found
[1299160.060381] hub 1-1.3:1.0: 4 ports detected
[1304677.285799] usb 1-1.3: USB disconnect, device number 25
[1307145.422526] usb 1-1.3: new high-speed USB device number 26 using ehci-pci
[1307145.514952] usb 1-1.3: New USB device found, idVendor=0424, idProduct=2504
[1307145.514957] usb 1-1.3: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[1307145.515208] hub 1-1.3:1.0: USB hub found
[1307145.515314] hub 1-1.3:1.0: 4 ports detected
[1310078.211196] usb 1-1.2: new full-speed USB device number 27 using ehci-pci
[1310078.305489] usb 1-1.2: New USB device found, idVendor=2341, idProduct=003d
[1310078.305494] usb 1-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=220
[1310078.305497] usb 1-1.2: Product: Arduino Due Prog. Port
[1310078.305499] usb 1-1.2: Manufacturer: Arduino (www.arduino.cc)
[1310078.305502] usb 1-1.2: SerialNumber: 85235353037351F081F0
[1310078.305904] cdc_acm 1-1.2:1.0: ttyACM0: USB ACM device
[38;5;202m[38;5;245mantodom[00m@[38;5;172mdis172ac[00m:[38;5;5m~[00mâŸ« lsusb
Bus 002 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 004: ID 046d:c31c Logitech, Inc. Keyboard K120 for Business
Bus 001 Device 003: ID 046d:c05b Logitech, Inc. M-U0004 810-001317 [B110 Optical USB Mouse]
Bus 001 Device 026: ID 0424:2504 Standard Microsystems Corp. USB 2.0 Hub
Bus 001 Device 027: ID 2341:003d Arduino SA 
Bus 001 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 002: ID 13fd:1340 Initio Corporation Hi-Speed USB to SATA Bridge
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
[38;5;202m[38;5;245mantodom[00m@[38;5;172mdis172ac[00m:[38;5;5m~[00mâŸ« re[K[Kuan [K[K[Kname -a
Linux dis172ac 3.13.0-32-generic #57-Ubuntu SMP Tue Jul 15 03:51:08 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
[38;5;202m[38;5;245mantodom[00m@[38;5;172mdis172ac[00m:[38;5;5m~[00mâŸ« lsb_release --all
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.1 LTS
Release:	14.04
Codename:	trusty
[38;5;202m[38;5;245mantodom[00m@[38;5;172mdis172ac[00m:[38;5;5m~[00mâŸ« dpkg -l | grep usb
ii  libg[01;38;5;202m[Kusb[m[K2:amd64                              0.1.6-5                                             amd64        GLib wrapper around lib[01;38;5;202m[Kusb[m[K1
ii  lib[01;38;5;202m[Kusb[m[K-0.1-4:amd64                          2:0.1.12-23.3ubuntu1                                amd64        userspace USB programming library
ii  lib[01;38;5;202m[Kusb[m[K-1.0-0:amd64                          2:1.0.17-1ubuntu2                                   amd64        userspace USB programming library
ii  lib[01;38;5;202m[Kusb[m[K-1.0-0:i386                           2:1.0.17-1ubuntu2                                   i386         userspace USB programming library
ii  lib[01;38;5;202m[Kusb[m[K-1.0-0-dbg:amd64                      2:1.0.17-1ubuntu2                                   amd64        userspace USB programming library development files
ii  lib[01;38;5;202m[Kusb[m[K-1.0-0-dev:amd64                      2:1.0.17-1ubuntu2                                   amd64        userspace USB programming library development files
ii  lib[01;38;5;202m[Kusb[m[K-1.0-doc                              2:1.0.17-1ubuntu2                                   all          documentation for userspace USB programming
ii  lib[01;38;5;202m[Kusb[m[K-dev                                  2:0.1.12-23.3ubuntu1                                amd64        userspace USB programming library development files
ii  lib[01;38;5;202m[Kusb[m[Kmuxd2                                 1.0.8-2ubuntu1                                      amd64        USB multiplexor daemon for iPhone and iPod Touch devices - library
ii  lib[01;38;5;202m[Kusb[m[Kredirhost1:amd64                      0.6-2ubuntu1                                        amd64        Implementing the [01;38;5;202m[Kusb[m[K-host (*) side of a [01;38;5;202m[Kusb[m[Kredir connection (runtime)
ii  lib[01;38;5;202m[Kusb[m[Kredirparser1:amd64                    0.6-2ubuntu1                                        amd64        Parser for the [01;38;5;202m[Kusb[m[Kredir protocol (runtime)
ii  spice-client-glib-[01;38;5;202m[Kusb[m[K-acl-helper            0.22-0nocelt2                                       amd64        Spice client glib [01;38;5;202m[Kusb[m[K acl helper
ii  [01;38;5;202m[Kusb[m[K-modeswitch                              2.1.1+repack0-1ubuntu1                              amd64        mode switching tool for controlling "flip flop" USB devices
ii  [01;38;5;202m[Kusb[m[K-modeswitch-data                         20140327-1                                          all          mode switching data for [01;38;5;202m[Kusb[m[K-modeswitch
ii  [01;38;5;202m[Kusb[m[Kmuxd                                     1.0.8-2ubuntu1                                      amd64        USB multiplexor daemon for iPhone and iPod Touch devices
ii  [01;38;5;202m[Kusb[m[Kutils                                    1:007-2ubuntu1                                      amd64        Linux USB utilities
ii  xserver-xorg-video-sis[01;38;5;202m[Kusb[m[K                   1:0.9.6-2build1                                     amd64        X.Org X server -- SiS USB display driver
[38;5;202m[38;5;245mantodom[00m@[38;5;172mdis172ac[00m:[38;5;5m~[00mâŸ« exit
exit

Script terminado (miÃ© 07 oct 2015 12:25:59 WEST
)

------- For the PC where bossac DOES NOT work ---------------------------------------------------------------
Script iniciado (mié 07 oct 2015 12:33:03 WEST
)antodom@dis172acp:~$ lsmod
Module                  Size  Used by
cdc_acm                28803  0 
uvcvideo               80885  0 
btrfs                 835994  0 
raid6_pq               97812  1 btrfs
xor                    21411  1 btrfs
ufs                    74890  0 
qnx4                   13317  0 
hfsplus               107516  0 
hfs                    54677  0 
minix                  36140  0 
ntfs                   97369  0 
msdos                  17332  0 
jfs                   181348  0 
xfs                   916488  0 
libcrc32c              12644  2 xfs,btrfs
bbswitch               13943  0 
rfcomm                 69160  0 
bnep                   19624  2 
bluetooth             391136  10 bnep,rfcomm
binfmt_misc            17468  1 
snd_hda_codec_hdmi     46368  1 
snd_hda_codec_realtek    65812  1 
videobuf2_vmalloc      13216  1 uvcvideo
videobuf2_memops       13362  1 videobuf2_vmalloc
videobuf2_core         40664  1 uvcvideo
videodev              134688  2 uvcvideo,videobuf2_core
snd_hda_intel          56531  3 
snd_hda_codec         193017  3 snd_hda_codec_realtek,snd_hda_codec_hdmi,snd_hda_intel
snd_hwdep              13602  1 snd_hda_codec
snd_pcm               102099  3 snd_hda_codec_hdmi,snd_hda_codec,snd_hda_intel
snd_page_alloc         18710  2 snd_pcm,snd_hda_intel
snd_seq_midi           13324  0 
snd_seq_midi_event     14899  1 snd_seq_midi
snd_rawmidi            30144  1 snd_seq_midi
intel_rapl             18773  0 
x86_pkg_temp_thermal    14205  0 
intel_powerclamp       14705  0 
coretemp               13435  0 
kvm_intel             143187  0 
kvm                   455843  1 kvm_intel
crct10dif_pclmul       14289  0 
snd_seq                61560  2 snd_seq_midi_event,snd_seq_midi
crc32_pclmul           13113  0 
i915                  788212  4 
ghash_clmulni_intel    13216  0 
wl                   6367819  0 
cryptd                 20359  1 ghash_clmulni_intel
snd_seq_device         14497  3 snd_seq,snd_rawmidi,snd_seq_midi
cfg80211              484040  1 wl
snd_timer              29482  2 snd_pcm,snd_seq
drm_kms_helper         55071  1 i915
mei_me                 18627  0 
serio_raw              13462  0 
drm                   303102  5 i915,drm_kms_helper
snd                    69322  17 snd_hda_codec_realtek,snd_hwdep,snd_timer,snd_hda_codec_hdmi,snd_pcm,snd_seq,snd_rawmidi,snd_hda_codec,snd_hda_intel,snd_seq_device,snd_seq_midi
lpc_ich                21080  0 
mei                    82276  1 mei_me
soundcore              12680  1 snd
i2c_algo_bit           13413  1 i915
shpchp                 37032  0 
joydev                 17381  0 
acer_wmi               32522  0 
mxm_wmi                13021  0 
sparse_keymap          13948  1 acer_wmi
video                  19476  2 i915,acer_wmi
parport_pc             32701  0 
mac_hid                13205  0 
wmi                    19177  2 acer_wmi,mxm_wmi
ppdev                  17671  0 
lp                     17759  0 
parport                42348  3 lp,ppdev,parport_pc
tg3                   166478  0 
psmouse               106692  0 
ahci                   34091  2 
sdhci_pci              23172  0 
ptp                    18933  1 tg3
libahci                32716  1 ahci
sdhci                  43015  1 sdhci_pci
pps_core               19382  1 ptp
antodom@dis172acp:~$ dmesg -[K| grep -[K[K[K[K[K[Ktail -n 25
[46151.455441] cfg80211:   (start_freq - end_freq @ bandwidth), (max_antenna_gain, max_eirp)
[46151.455443] cfg80211:   (2402000 KHz - 2472000 KHz @ 40000 KHz), (300 mBi, 2000 mBm)
[46151.455444] cfg80211:   (2457000 KHz - 2482000 KHz @ 40000 KHz), (300 mBi, 2000 mBm)
[46151.455446] cfg80211:   (2474000 KHz - 2494000 KHz @ 20000 KHz), (300 mBi, 2000 mBm)
[46151.455447] cfg80211:   (5170000 KHz - 5250000 KHz @ 40000 KHz), (300 mBi, 2000 mBm)
[46151.455448] cfg80211:   (5735000 KHz - 5835000 KHz @ 40000 KHz), (300 mBi, 2000 mBm)
[46151.447435] Restarting tasks ... done.
[46151.471173] [drm] Enabling RC6 states: RC6 on, RC6p off, RC6pp off
[46151.779012] video LNXVIDEO:00: Restoring backlight state
[46151.779024] video LNXVIDEO:01: Restoring backlight state
[46151.779046] bbswitch: disabling discrete graphics
[46151.779064] ACPI Warning: \_SB_.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20131115/nsarguments-95)
[46151.961444] uvcvideo: Found UVC 1.00 device 1.3M WebCam (04fc:2800)
[46151.978934] input: 1.3M WebCam as /devices/pci0000:00/0000:00:1a.0/usb1/1-1/1-1.3/1-1.3:1.0/input/input17
[46151.979162] usbcore: registered new interface driver uvcvideo
[46151.979167] USB Video Class driver (1.1.1)
[48919.175124] usb 2-1.2: new full-speed USB device number 3 using ehci-pci
[48919.269762] usb 2-1.2: New USB device found, idVendor=2341, idProduct=003d
[48919.269774] usb 2-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=220
[48919.269780] usb 2-1.2: Product: Arduino Due Prog. Port
[48919.269785] usb 2-1.2: Manufacturer: Arduino (www.arduino.cc)
[48919.269790] usb 2-1.2: SerialNumber: 85235353037351F081F0
[48919.602226] cdc_acm 2-1.2:1.0: ttyACM0: USB ACM device
[48919.602615] usbcore: registered new interface driver cdc_acm
[48919.602617] cdc_acm: USB Abstract Control Model driver for USB modems and ISDN adapters
antodom@dis172acp:~$ lsusb
Bus 002 Device 003: ID 2341:003d Arduino SA 
Bus 002 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 003: ID 04fc:2800 Sunplus Technology Co., Ltd 
Bus 001 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
antodom@dis172acp:~$ uname -a
Linux dis172acp 3.13.0-65-generic #105-Ubuntu SMP Mon Sep 21 18:50:58 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
antodom@dis172acp:~$ lsb_release --all
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.3 LTS
Release:	14.04
Codename:	trusty
antodom@dis172acp:~$ dpkg -i[Kl | grep usb
ii  cups-filters-ipp[01;31m[Kusb[m[Kxd                                 1.0.52-0ubuntu1.3                                   amd64        ipp[01;31m[Kusb[m[Kxd - Daemon for IPP-over-USB printer support
ii  libg[01;31m[Kusb[m[K2:amd64                                        0.1.6-5                                             amd64        GLib wrapper around lib[01;31m[Kusb[m[K1
ii  lib[01;31m[Kusb[m[K-0.1-4:amd64                                    2:0.1.12-23.3ubuntu1                                amd64        userspace USB programming library
ii  lib[01;31m[Kusb[m[K-1.0-0:amd64                                    2:1.0.17-1ubuntu2                                   amd64        userspace USB programming library
ii  lib[01;31m[Kusb[m[K-1.0-0:i386                                     2:1.0.17-1ubuntu2                                   i386         userspace USB programming library
ii  lib[01;31m[Kusb[m[K-1.0-0-dev:amd64                                2:1.0.17-1ubuntu2                                   amd64        userspace USB programming library development files
ii  lib[01;31m[Kusb[m[K-1.0-doc                                        2:1.0.17-1ubuntu2                                   all          documentation for userspace USB programming
ii  lib[01;31m[Kusb[m[K-dev                                            2:0.1.12-23.3ubuntu1                                amd64        userspace USB programming library development files
ii  lib[01;31m[Kusb[m[Kmuxd2                                           1.0.8-2ubuntu1                                      amd64        USB multiplexor daemon for iPhone and iPod Touch devices - library
ii  [01;31m[Kusb[m[K-creator-common                                    0.2.56.3ubuntu0.2                                   amd64        create a startup disk using a CD or disc image (common files)
ii  [01;31m[Kusb[m[K-creator-gtk                                       0.2.56.3ubuntu0.2                                   amd64        create a startup disk using a CD or disc image (for GNOME)
ii  [01;31m[Kusb[m[K-modeswitch                                        2.1.1+repack0-1ubuntu1                              amd64        mode switching tool for controlling "flip flop" USB devices
ii  [01;31m[Kusb[m[K-modeswitch-data                                   20140327-1                                          all          mode switching data for [01;31m[Kusb[m[K-modeswitch
ii  [01;31m[Kusb[m[Kmuxd                                               1.0.8-2ubuntu1                                      amd64        USB multiplexor daemon for iPhone and iPod Touch devices
ii  [01;31m[Kusb[m[Kutils                                              1:007-2ubuntu1                                      amd64        Linux USB utilities
ii  xserver-xorg-video-sis[01;31m[Kusb[m[K                             1:0.9.6-2build1                                     amd64        X.Org X server -- SiS USB display driver

**Comment by Atirma** [comment]
Hi, 
Just a quick update/hint on this issue. We had tested that everything is fine until kernel 3.13.0-64-generic is installed (i.e. 3.13.0-63-generic is the last kernel where this problem does not show up). 

We have tested this booting the system through the "Advanced options for Ubuntu" in the grub booting screen and selecting the adequate kernel image. This is also an EMERGENCY shortcut to deal with this problem.

This issue is clearly related to reading data from a USB device. While dealing with these tests, we found also an unexpected problem  with a USB XBee adapter where outgoing messages were sent without problem but incoming messages were not received (in esence the same problem with bossac). This problem was solved also if a 3.13.0-63-generic (or previous) kernel was used.

Regards

**Comment by NicoHood** [comment]
What usb port are you using? The one connected to the DUE or the one connected to the 16u2.
If its the 2nd, then its probably a delay problem. the chip erase and reset timing might be wrong due to a lagg. USB2 or 3 used here? with USB HUB and which version (2.0 HUB or 3.0)?

**Comment by Atirma** [comment]
Hi, 
This problem occurs independently of the port used. They were connected to a USB 2.0 port.
Regards

**Comment by NicoHood** [comment]
You mean on the DUE USB port and the 16u2 USB Port? (you know your arduino has 2 usb plugs).
Have you tested something like an uno or leonardo too?

**Comment by Atirma** [comment]
Yes, we found this problem with the two DUE's port (the programming port and the native port). However, we haven't tested other boards yet.

**Comment by NicoHood** [comment]
Other boards would be nice to test. I only have AVR boards, but I programmed/modified a lot of the bootloaders. So I could play with this under my linux (where everything runs fine though).

**Comment by facchinm** [comment]
The changelog between 3.13.0-63 and 3.13.0-64 (https://launchpad.net/ubuntu/+source/linux/3.13.0-64.104) shows that a bunch of modifications has been made for both the tty and the usb subsystems.
Particularly, I'm afraid that this two commits can be related with the problem:
```
  * xhci: Calculate old endpoints correctly on device reset
    - LP: #1493305
  * n_tty: Refactor input_available_p() by call site
    - LP: #1397976
```
However we can only report this issue to Ubuntu maintainers as a generic bug until we are able to bisect the offending commit

**Comment by facchinm** [comment]
We are not alone :smile: https://bugreports.qt.io/browse/QTBUG-48499

**Comment by facchinm** [comment]
Reported [here] (https://bugs.launchpad.net/ubuntu/+source/linux-lts-trusty/+bug/1500860)

**Comment by matthijskooijman** [comment]
I'm running a Debian machine with kernel 4.0.2, which can upload to the Due (both ports) just fine (just tested with Arduino IDE 1.6.5). Perhaps this is a kernel bug that has since been fixed and we shouldn't spend to much time on this? Perhaps the original reporter can try a newer kernel version (from a newer Ubuntu version?) and see if that helps?

**Comment by facchinm** [comment]
@matthijskooijman , I'm running kernel 4.2 on Arch so I should be even less concerned :smile: 
The problem here is that 3.13 kernel has been chosen by Ubuntu as LTS kernel for 14.04 (so support will end in 2017) and ubuntu.com/download/desktop points to that release...

**Comment by NicoHood** [comment]
I am running kernel:
Linux linuxpc 3.16.0-50-generic #67~14.04.1-Ubuntu SMP Fri Oct 2 22:07:51 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

I have no due, but all other avr devices work. Using Elementary OS based on Ubuntu 14 LTS

**Comment by Atirma** [comment]
Hi,
Good news. It looks like this problem is solved in  the latest kernel (3.13.0-66-generic) update. 
Regards

**Comment by facchinm** [comment]
:+1: :+1: 
https://launchpad.net/ubuntu/+source/linux/3.13.0-66.107 shows it was solved by
* n_tty: Fix poll() when TIME_CHAR and MIN_CHAR == 0
    - LP: #1397976
